bridge_adaptivity/module/views.py

```diff
@@ -45,18 +43,19 @@ class GetGradingPolicyForm(FormView):
     form_class = GradingPolicyForm
     template_name = 'module/gradingpolicy_form.html'
     prefix = 'grading'
-    # grading_policy_name: FormClass
-    grading_policy_forms_map = {}
 
     def get_form_class(self):
-        return self.grading_policy_forms_map.get(self.request.GET.get('grading_policy'), self.form_class)
+        policy_cls = GRADING_POLICY_NAME_TO_CLS.get(self.request.GET.get('grading_policy'), None)
+        if policy_cls is None:
+            raise Http404("No such grading policy")
+        return policy_cls.get_form_class()
 
     def get_form(self, form_class=None):
         form = super(GetGradingPolicyForm, self).get_form()
         if self.kwargs.get('group_slug'):
             get_object_or_404(CollectionGroup, slug=self.kwargs['group_slug'])
         gp = self.request.GET.get('grading_policy')
-        if gp and gp in VALID_GRADING_POLICIES:
+        if gp and gp in GRADING_POLICY_NAME_TO_CLS.keys():
```

Comment:
We could simplify this statement to: `if gp in GRADING_POLICY_NAME_TO_CLS` -- and it would be the same